### PR TITLE
Use salt bundle

### PR DESCRIPTION
**Pull request by nadvornik**
Use salt bundle in saltboot.
Saltboot now works with both bundle and normal minion. Default image configurations are changed to use bundle.

### CODE REVIEW

**Review by aaannz — COMMENTED** [review]
- About the `migrate-to-bundle`, how can such condition happen?
- What about migrating from bundle back to salt-minion. Seems like initrd can cope with it, but should we support that? What are your thoughts?
- profiles in https://github.com/SUSE/manager-build-profiles should be updated as well. IMO these ones can be dropped together with packages. Everyone is using git.

File: dracut-saltboot/saltboot/module-setup.sh [hunk 1]
```diff
@@ -51,9 +51,16 @@ installkernel() {
 
 # called by dracut
 install() {
-    inst_multiple -o $(rpm -ql $(get_python_pkg_deps_recursive salt salt-minion) | \
+    if rpm -q venv-salt-minion ; then
+        export LD_LIBRARY_PATH=/usr/lib/venv-salt-minion/lib
+        inst_multiple -o $(rpm -ql venv-salt-minion | \
                   grep -v '\.pyc$\|/etc/salt/minion_id\|/etc/salt/pki\|/usr/share/doc/\|/usr/share/man' | \
                   fix_python_deps )
```
[the remark this answers is not in the document]
Comment (nadvornik) [reply dracut-saltboot/saltboot/module-setup.sh:58]: I think so. The bundle contains non-executable .so files too.

File: dracut-saltboot/saltboot/saltboot.sh [hunk 2]
```diff
@@ -280,27 +292,34 @@ if [ "$systemIntegrity" = "unknown" ] ; then
...
-cat > /etc/salt/minion.d/grains-initrd.conf <<EOT
+cat > $INITRD_SALT_ETC/minion.d/grains-initrd.conf <<EOT
 grains:
   saltboot_initrd: False
 EOT
 
-rm -f /etc/salt/minion.d/autosign-grains-onetime.conf
+rm -f $INITRD_SALT_ETC/minion.d/autosign-grains-onetime.conf
+
+if test -e $NEWROOT/etc/venv-salt-minion ; then
```
[the remark this answers is not in the document]
Comment (nadvornik) [reply dracut-saltboot/saltboot/saltboot.sh]: fixed

File: dracut-saltboot/dracut-saltboot.spec [hunk 3]
```diff
@@ -34,6 +34,7 @@ Requires:       curl
 Requires:       device-mapper
 Requires:       parted
 Requires:       salt-minion
```
[the remark this answers is not in the document]
Comment (nadvornik) [reply dracut-saltboot/dracut-saltboot.spec]: There is no common string, I have removed this line.
Also, I changed module-setup to hard-fail if all minion packages are missing.

**Comment by nadvornik** [comment]
> * About the `migrate-to-bundle`, how can such condition happen?

Image is deployed by old initrd and there is no reboot because of kernel version.

> 
>     * What about migrating from bundle back to salt-minion. Seems like initrd can cope with it, but should we support that? What are your thoughts?
> 

We should support all combinations, it is better than debug it as L3.
I can imagine there is a mix of old images with the new ones.

**Comment by nadvornik** [comment]
I have sorted out mandatory binaries and let module-setup fail if sometging is missing.
Also, I moved the image srvices from image sources to dracut-saltboot package.

**Review by aaannz — APPROVED** [review]

**Pull request merged**